File: bench/prime.pl
use v5;
use strict;

my $maxnum = 100000;

sub is_prime {
    my $n = shift;
    my $j = 2;
    while ($j * $j < $n + 1) {
        if ($n % $j == 0) {
            return 0;
        }
        $j++;
    }
    return 1;
}

for (my $i = 2; $i <= $maxnum; $i++) {
    if (is_prime($i)) {
        print($i, "\n");
    }
}

File: bench/prime.rb
maxnum = 100000
i = 2

def is_prime(n)
    j = 2
    while j * j < n + 1
        if n % j == 0
            return false
        end
        j += 1
    end
    return true
end


while i <= maxnum
    p i if is_prime(i)
    i += 1
end

